client/lib/infrastructure/auth/api_auth_repository.dart

```diff
@@ -68,4 +69,47 @@ class ApiAuthRepository implements IAuthRepository {
       return left(const AuthFailure.networkError());
     }
   }
+
+  @override
+  Future<Option<User>> getCachedUser() async {
+    try {
+      final SharedPreferences prefs = await SharedPreferences.getInstance();
+
+      final String? response = prefs.getString("user");
+      
+      final UserDto userDtoIn =
+          UserDto.fromJson(jsonDecode(response!) as Map<String, dynamic>);
+      return optionOf(userDtoIn.toDomain());
+    } catch (err) {
+      return none();
+    }
+  }
+
+  @override
+  Future<Either<AuthFailure, Unit>> setCachedUser({required User user}) async {
+    try {
+      final SharedPreferences prefs = await SharedPreferences.getInstance();
+
+      final UserDto userDtoOut = UserDto.fromDomain(user);
+
+      prefs.setString("user", jsonEncode(userDtoOut.toJson()));
+
+      return right(unit);
+    } catch (err) {
+      return left(const AuthFailure.serverError());
```

Comment:
Better you use another failure dont you think?
 makes it easier to handle

Reply:
You're right. Was using databaseError but some refactoring required for UI. Since I dont have @MichaelAlemayehu1 register UI in this branch I figured I will do it later.